iron-core/src/main/java/io/axway/iron/core/internal/StoreManagerFactoryBuilderConfigurator.java

```diff
@@ -0,0 +1,201 @@
+                            LOG.warn("A {} is already configured (with {})", type.getTypeName(), map.get(type).supplierClass().getName());
+                        }
+                        BuilderImplConfig builderImplConfig = new BuilderImplConfig(clazz, propertyName);
+                        LOG.info("Adding the configured class in the builder list: {{}: {}}", type.getTypeName(), className);
+                        map.put(type, builderImplConfig);
+                    }
+                } catch (ClassNotFoundException e) {
+                    // nothing to do
+                }
+            }
+        }
+        return map;
+    }
+
+    private Supplier<?> instantiateBuilder(Class<Supplier<?>> componentBuilderClazz, Properties properties, String baseName) {
+        try {
+            Constructor<Supplier<?>> constructor = componentBuilderClazz.getConstructor();
+            Supplier<?> builder = constructor.newInstance();
+
+            for (Method method : componentBuilderClazz.getDeclaredMethods()) {
+                if (method.getName().startsWith("set")) {
+                    Class<?>[] parameterTypes = method.getParameterTypes();
+                    if (parameterTypes.length == 1) {
+                        String propertyName = method.getName().substring(3);
+                        Object value = getPropertyValue(parameterTypes[0], properties, baseName, propertyName);
+                        if (value != null) { // do not call setter if value is null
+                            method.invoke(builder, value);
+                        } else if (isMandatory(method)) {
+                            throw new ConfigurationException("Parameter '" + propertyName + "' for '"+ componentBuilderClazz.getClass().getCanonicalName()+"' is mandatory!");
+                        }
+                    } else {
+                        LOG.error("The configured builder has setters with more than one parameter {{}.{}}", componentBuilderClazz.getName(), method.getName());
+                    }
+                }
+            }
+            return builder;
+        } catch (NoSuchMethodException | InvocationTargetException | InstantiationException | IllegalAccessException e) {
+            throw new ConfigurationException(e);
+        }
+    }
+
+    private boolean isMandatory(Method method) {
+        return stream(method.getParameterAnnotations()[0]).noneMatch(x -> Nullable.class.isAssignableFrom(x.annotationType()));
+    }
+
+    @Nullable
+    private <T> T getPropertyValue(Class<T> clazz, Properties properties, String baseName, String propertyName) {
+        if (clazz.isAssignableFrom(Properties.class)) {
+            return extractProperties(clazz, properties, baseName);
+        } else {
+            String fullName = baseName + "." + camelCaseToLowerCaseUnderscore(propertyName);
+            if (clazz.isAssignableFrom(Path.class)) {
+                return extractPath(clazz, properties, fullName);
```

Comment:
you can factorize out of individual if branch/extract methods:
* the getProperty
* the clazz.cast

Reply:
ok.
I factorized the getProperty, and completely removed the clazz.cast